code/modules/projectiles/guns/energy/energy_gun.dm

```diff
@@ -227,3 +227,10 @@
 			D.obj_flags &= ~EMAGGED
 		investigate_log("dna lock reset by [key_name(user)]", INVESTIGATE_RECORDS)
 	..()
+
+/obj/item/gun/energy/smartpistol/emp_act(severity)
+	audible_message("<span class='italics'>You hear erratic beeping from \the [name].</span>", null,  1)
+	var/obj/item/firing_pin/dna/D = pin
+	D.unique_enzymes = null
```

Comment:
this will runtime if gun doesn't have pin

Reply:
If pin code is correct the pin can't be removed.

Reply:
imo better be safe